pkg/statedb/txn.go

```diff
@@ -225,7 +226,7 @@ func (txn *txn) Insert(meta TableMeta, guardRevision Revision, data any) (any, b
 				}
 			})
 		}
-		newKeys.Foreach(func(newKey []byte) {
+		newKeys.Foreach(func(newKey index.Key) {
 			// Non-unique secondary indexes are formed by concatenating them
 			// with the primary key.
 			if !indexer.unique {
```

Comment:
Now that I think about this secondary index construction, I think there could have been issues when combining a variable length unique index with a variable length non-unique index.
Let's say we have a unique index name, and a non-unique index display name, both strings

if we had the the two objects `{name: "bar", displayName: "foo"}` and `{name: "obar", displayName: "fo"}`, what would a displayname query to `foobar` have resulted in?

Reply:
Without proper termination or fixed-size keys, then indeed this would've resulted in matching the second object. With termination of strings we'd have `fo\0obar\0` as the key and thus querying `foobar` would prefix seek to `foobar\0` and not match.